### PR TITLE
Particle system javadocs

### PR DESCRIPTION
**Pull request by MaxBorsch**
Particle System javadocs and new big TODO for GCI.

### CODE REVIEW

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

File: configs/engine/rendering.cfg [hunk 1]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (vampcat) [inline configs/engine/rendering.cfg]: Unstage this file. In fact, if you don't mind, add it to gitignore.

(this file is a side-effect of a PR of mine that got merged recently, and i forgot to gitignore this)

File: facade.groovy [hunk 2]
```diff
@@ -0,0 +1,213 @@
+// We use GrGit for interacting with Git. This gets a hold of it as a dependency like Gradle would
```
Comment (vampcat) [inline facade.groovy]: I'm not sure why you are commiting this file in a ParticleSystem Javadocs PR.

**Comment by MaxBorsch** [comment]
@vampcat Fixed. I don't think manuel3d needs to look at this, I wrote all of the particle system code last year that I just javadoc'ed.

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by vampcat** [comment]
I know that you wrote the particle system code for last gci. However, since I want a second pair of eyes that know what they are looking at - i figured manu, who was (afaik) involved in that PR - would be a good candidate.

**Comment by MaxBorsch** [comment]
@vampcat Ah ok. Is it possible to get my task approved though so I can submit my next few? Unless manu will be on some time soon.

**Comment by vampcat** [comment]
I'll have one more look at the javadocs, and approve the task. We can let the pr stick around for 2-3 days, in case someone else wants to review.

File: engine/src/main/java/org/terasology/particles/ParticleSystemManager.java [hunk 3]
```diff
@@ -24,17 +24,35 @@
...
 
+    /**
+     * Registers an Affector function to be used when updating particle systems that have the registered affector as a component.
+     *
+     * @param affectorFunction The affector function to be used.
+     */
     void registerAffectorFunction(AffectorFunction affectorFunction);
 
+    /**
+     * Registers an Affector function to be used when updating particle systems that have the registered affector as a component.
```
Comment (vampcat) [inline engine/src/main/java/org/terasology/particles/ParticleSystemManager.java]: Registers an -affector- function? Either I'm missing something, or you did not update the description after copy-pasting it.

File: engine/src/main/java/org/terasology/particles/updating/ParticleUpdater.java [hunk 4]
```diff
@@ -41,18 +41,34 @@
     void dispose(EntityRef entity);
 
     /**
-     * @param emitter                    The particle emitter that is being updated.
-     * @param registeredAffectorFunctions  The list of affector functions to use when processing the given system's affectors.
+     * @param emitter The particle emitter that is being updated.
```
Comment (vampcat) [inline engine/src/main/java/org/terasology/particles/updating/ParticleUpdater.java]: Any reason you did not add a description for this method?

Comment (MaxBorsch) [reply engine/src/main/java/org/terasology/particles/updating/ParticleUpdater.java]: Oh I missed that, those javadocs were already there from my original code last year, I didn't notice that they were incomplete. I'll go check them, thanks for pointing that out. @vampcat

**Comment by MaxBorsch** [comment]
@vampcat Fixed and resubmitted task.

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

File: .gitignore [hunk 5]
```diff
@@ -92,6 +92,6 @@ nuiEditorAutosave.json
 # Ignore stuff we extract for use with Terasology
 /natives/
 config/metrics/
-
+configs/engine/rendering.cfg
```
Comment (oniatus) [inline .gitignore]: We have a similar change at #3227 and this is not related to javadoc.
Can we exclude it here and have a separate PR for excluding files from the FlexibleConfig?

File: engine/src/main/java/org/terasology/particles/ParticleSystemManagerImpl.java [hunk 6]
```diff
@@ -91,7 +93,10 @@ public void onEmitterDeactivated(BeforeDeactivateComponent event, EntityRef enti
         particleUpdater.dispose(entity);
     }
 
-
+    // TODO: Make generator / affector functions be maintained by a particle updater and register automatically when used by a new emitter.
```
Comment (oniatus) [inline engine/src/main/java/org/terasology/particles/ParticleSystemManagerImpl.java]: This should be an issue and not TODO hidden somewhere in the javadocs.
